course04/homework00/ciaranfrain/sorted_array_tests.cpp

```diff
@@ -0,0 +1,132 @@
+#include "sorted_array_tests.h"
+
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+const int SortedArrayTests::mDefaultMaxSize = 10;
+
+SortedArrayTests::SortedArrayTests(): mSortedArray(mDefaultMaxSize)
+{}
+
+void SortedArrayTests::RunTests()
+{
+	std::cout << "Running tests..." << std::endl;
+
+	RunTest("ShouldInitialiseEmpty",             std::bind(&SortedArrayTests::ShouldInitialiseEmpty, this));
+	RunTest("ShouldInsertOneElement",            std::bind(&SortedArrayTests::ShouldInsertOneElement, this));
+	RunTest("ShouldRetrieveCorrectElementValue", std::bind(&SortedArrayTests::ShouldRetrieveCorrectElementValue, this));
+	RunTest("ShouldThrowOnExceedingSize",        std::bind(&SortedArrayTests::ShouldThrowOnExceedingSize, this));
+	RunTest("ShouldInsertLesserValueAtFront",    std::bind(&SortedArrayTests::ShouldInsertLesserValueAtFront, this));
+	RunTest("ShouldInsertGreaterValueAtBack",    std::bind(&SortedArrayTests::ShouldInsertGreaterValueAtBack, this));
+	RunTest("ShouldSortElements",                std::bind(&SortedArrayTests::ShouldSortElements, this));
+}
+
+void SortedArrayTests::Reset(std::size_t size = mDefaultMaxSize)
+{
+	mSortedArray = SortedArray(size);
```

Comment:
What would have been even better than this Reset method, is to construct the test object that contains this array. It's called a fixture - we will talk about it tomorrow. Why this is better: here, you rely on a correct implementation of the assignment op... It's usually better to start every unit test with a freshly constructed object